src/test/groovy/com/github/benmanes/gradle/versions/DependencyUpdatesSpec.groovy

```diff
@@ -636,28 +579,84 @@ final class DependencyUpdatesSpec extends Specification {
     }
   }
 
-  private def addBadRepositoryTo(project) {
+  private static void addBadRepositoryTo(project) {
     project.repositories {
       maven { url = 'http://www.example.com' }
     }
   }
 
-  private def addDependenciesTo(project) {
+  private static void addDependenciesTo(project) {
     project.configurations {
       upToDate
       exceedLatest
       upgradesFound
       unresolvable
     }
     project.dependencies {
-      upToDate 'backport-util-concurrent:backport-util-concurrent:3.1',
-        'backport-util-concurrent:backport-util-concurrent-java12:3.1'
-      exceedLatest 'com.google.guava:guava:99.0-SNAPSHOT',
-        'com.google.guava:guava-tests:99.0-SNAPSHOT'
-      upgradesFound 'com.google.inject:guice:2.0',
-        'com.google.inject.extensions:guice-multibindings:2.0'
-      unresolvable 'com.github.ben-manes:unresolvable:1.0',
-        'com.github.ben-manes:unresolvable2:1.0'
+      upToDate('backport-util-concurrent:backport-util-concurrent:3.1') { because 'I said so' }
+      upToDate('backport-util-concurrent:backport-util-concurrent-java12:3.1')
+      exceedLatest('com.google.guava:guava:99.0-SNAPSHOT') { because 'I know the future' }
+      exceedLatest('com.google.guava:guava-tests:99.0-SNAPSHOT')
+      upgradesFound('com.google.inject:guice:2.0') { because 'That\'s just the way it is' }
+      upgradesFound('com.google.inject.extensions:guice-multibindings:2.0')
+      unresolvable('com.github.ben-manes:unresolvable:1.0') { because 'Life is hard' }
+      unresolvable('com.github.ben-manes:unresolvable2:1.0')
```

Comment:
In order to provide "reason" entries, I had to declare the dependencies slightly differently. Rather than only update the ones I needed to, I changed them all for consistency.